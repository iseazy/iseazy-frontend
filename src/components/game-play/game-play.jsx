import { GameCard } from '../game-card';

export const GamePlay = ({ cards, onClick }) => (
    <div className="game-play">
        <div className="game-play__grid">
            { cards.length ? cards.map(({ id, src, flipped }, index) => (
                <GameCard key={ `card-${ id }`}
                    id={ id }
                    src={ src }
                    flipped={ flipped }
                    title={ index + 1 }
                    onClick={ onClick } />
            )) : null }
        </div>
    </div>
);
