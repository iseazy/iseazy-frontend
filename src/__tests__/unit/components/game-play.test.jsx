import { render } from '@testing-library/react';
import { GamePlay } from '../../../components/game-play/game-play';

jest.mock('../../../components/game-card/game-card', () => ({
    GameCard: ({ onClick, ...props }) => `card with props [${Object.entries(props).map(([a,b]) => `${ a }=${ b }`).join(';')}]\n`
}));

describe('<GamePlay />', () => {    
    it('should render component with cards', () => {
        const testProps = {
            cards: mockDeck,
            onClick: () => {}
        }
        const { asFragment } = render(<GamePlay { ...testProps }/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render component without cards', () => {
        const testProps = {
            cards: [],
            onClick: () => {}
        }
        const { asFragment } = render(<GamePlay { ...testProps }/>);
        expect(asFragment()).toMatchSnapshot();
    });
});