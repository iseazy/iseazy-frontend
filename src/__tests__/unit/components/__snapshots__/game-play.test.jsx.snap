// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamePlay /> should render component with cards 1`] = `
<DocumentFragment>
  <div
    class="game-play"
  >
    <div
      class="game-play__grid"
    >
      card with props [id=1;src=testSourceA;flipped=true;title=1]
card with props [id=2;src=testSourceA;flipped=true;title=2]
card with props [id=3;src=testSourceB;flipped=true;title=3]
card with props [id=4;src=testSourceB;flipped=false;title=4]
card with props [id=5;src=testSourceC;flipped=false;title=5]
card with props [id=6;src=testSourceC;flipped=false;title=6]

    </div>
  </div>
</DocumentFragment>
`;

exports[`<GamePlay /> should render component without cards 1`] = `
<DocumentFragment>
  <div
    class="game-play"
  >
    <div
      class="game-play__grid"
    />
  </div>
</DocumentFragment>
`;
